.github/workflows/library-release.yml

```diff
@@ -1,23 +1,116 @@
-name: Build and Release
+name: Publish c2pa-library binaries
 
 on:
   push:
-    branches: [ "gpeacock/c2pa_c" ]
     tags:
       - 'c2pa-v*' # Trigger on version tags (e.g., v1.0.0)
   workflow_dispatch: # Allow manual triggering of the workflow
 
-env:
-  CARGO_TERM_COLOR: always
-
 jobs:
   build:
+    name: Publish c2pa-library binaries
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+        rust_version: [ stable ]
+        experimental: [ false ]
+        include:
+          - os: macos-latest
+            target: aarch64-apple-darwin
 
-    runs-on: ubuntu-latest
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
 
+      - name: Setup Strawberry Perl
+        if: matrix.os == 'windows-latest'
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: '5.32'
+          distribution: 'strawberry'
+      
+      # Install Locale::Maketext::Simple only on Windows with explicit paths
+      - name: Install Locale::Maketext::Simple and configure paths
+        if: matrix.os == 'windows-latest'
+        run: |
+          cpanm --verbose Locale::Maketext::Simple
+          # List installed modules to confirm installation
+          cpanm --info Locale::Maketext::Simple
+          # Display Perl's include paths
+          perl -e "print join(\"\n\", @INC);"
+          # Create a local copy in the expected path
+          perl -MLocale::Maketext::Simple -e "print \$INC{'Locale/Maketext/Simple.pm'}" > module_path.txt
+          $MODULE_PATH = Get-Content -Path module_path.txt
+          echo "Module found at: $MODULE_PATH"
+          
+          # Create symbolic links to the modules in paths that are being checked
+          # Set environment variable for subsequent steps
+          echo "PERL5LIB=$env:PERL5LIB;D:\a\c2pa-rs\c2pa-rs\target\x86_64-pc-windows-msvc\release\build\openssl-sys-def693d178636927\out\openssl-build\build\src\util\perl" | Out-File -FilePath $env:GITHUB_ENV -Append
+        shell: pwsh
+     
+      - name: Build release
+        run: make release
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.os }}-${{ matrix.target }}-release
+          path: target/artifacts/*
+
+  release:
+    needs: build
+    runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - name: Build
-      run: cargo build --verbose
-    - name: Run tests
-      run: cargo test --verbose
+      - name: Download build artifacts
+        uses: actions/download-artifact@v3
```

Comment:
Why not v4? The Python release steps already use v4 => https://github.com/contentauth/c2pa-python/blob/b41be67cd6bc9a23f879e554c7fa8dee070e75d2/.github/workflows/build.yml#L371